### PR TITLE
Pre-release merge for submission v0.1-alpha

### PR DESCRIPTION
**Pull request by erickpeirson**
**Review will close at 1pm on Tuesday, June 4.** Subsequent reviews will take place on an ongoing basis during alpha and beta testing.

This is a pre-alpha merge for v0.1 of the submission UI. The original goals for this release were:

- Provide all existing services of classic system (feature-parity), with seamless integration with other classic components (e.g. moderation, announcement). 
- Modest improvement to user experience. 
- Move submission UI to new unified design paradigm (color scheme, fonts). 
- Focus on reducing confusion, improving messaging to address problems related to frequent user support issues. 
- Under the hood, implement a new data architecture that better scales and evolves to meet user and stakeholder needs.

The latter objective is covered by https://github.com/arXiv/arxiv-submission-core/pull/64

Some of the major differences from the classic UI include:
- The submission process is broken down into smaller steps
- JREF, withdrawal, and cross-list requests begin to be treated differently than new submissions and replacements
- Highlighting of key terms/phrases in the TeX compilation log; this is the first incremental move toward a more user-friendly compilation experience (e.g. taxonomy of known errors, distilled log output, etc)
- HTML email confirmation  
- Native UTF8 support

Under the hood, we have broken out file management (upload), compilation, and plain text extraction as stand-alone services. See [the arXiv submission docs](https://arxiv.github.io/arxiv-submission-core/architecture.html#services-building-blocks) for an overview. 

Practically speaking, this means that:
- the UI is calling the file management and compilation services behind the scenes, and 
- the UI is no longer responsible for orchestrating text extraction and calling the autoclassifier -- this is now handled by the [submission agent](https://arxiv.github.io/arxiv-submission-core/architecture.html#submission-agent). So is sending the confirmation e-mail.

# How to review

Per the [arXiv NG release process](https://arxiv.github.io/arxiv-arxitecture/crosscutting/testing_and_release_process.html#alpha-does-it-work), **alpha** means that:

1. The majority of and/or highest priority functional goals for the release have been met.
2. Bugs are expected to exist, and known bugs may not yet have been fully addressed.
3. The release is deployed for internal review. This may be on the on-premises “beta” server, or in a staging environment in the cloud.
4. Dev and operations teams all participate in review, and report bugs, improvement requests, and other feedback to the feature team. A two-stage alpha review will first target the core operations team, and then broaden testing to volunteer moderators and close partners.
5. Feature team begins to prioritize and fix bugs, performs load testing, and works to evaluate the release against quality and performance goals.

The first round of alpha testing will involve some volunteers outside the core ops team.

**In advance of alpha**, therefore, please take a look at the following things:
- If you have not yet done so already, please review the [architecture for the arXiv-NG](https://arxiv.github.io/arxiv-submission-core/architecture.html) submission system.
- Try spinning up the UI and other submission services using docker-compose. See https://github.com/arXiv/arxiv-submission-ui#quick-start for instructions. 
- Please help to find any obvious bugs. Try running some announced source packages through, or take a look at the [filemanager tests](https://github.com/arXiv/arxiv-filemanager/tree/develop/tests) for lots of oddball submissions. The goal is to avoid anything terribly embarrassing when we enter alpha.

# Previews
## Submission dashboard
http://127.0.0.1:8000/
image

## Verify contact information
http://127.0.0.1:8000/1/verify_user
image

## Confirm authorship
http://127.0.0.1:8000/1/authorship
image

## Select a license
http://127.0.0.1:8000/1/license
image

## Accept submission policies
http://127.0.0.1:8000/1/policy
image

## Primary classification
http://127.0.0.1:8000/1/classification
image

## Cross-list categories
http://127.0.0.1:8000/1/cross_list
image

## File upload
http://127.0.0.1:8000/1/file_upload
image
image

## Compile PDF from source
http://127.0.0.1:8000/1/file_process
image
image
image

## Core metadata
http://127.0.0.1:8000/1/add_metadata
image

## Optional metadata
http://127.0.0.1:8000/1/add_optional_metadata
image

## Final preview

image

## Submission confirmation

image

## Confirmation email (via MailHog)
You can access mailhog at http://127.0.0.1:8025
image
image

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,240 @@
...
+
+```
+submission-bootstrap     | 1 picoline2058@gmail.com eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJzZXNzaW9uX2lkIjoiYWQyY2UxNmUtMjQwNi00NjgwLWI2NWItMDE3NGYyNDA0MzhlIiwic3RhcnRfdGltZSI6IjIwMTktMDItMTNUMTI6MjM6MzQuMjE2NzcxLTA1OjAwIiwidXNlciI6eyJ1c2VybmFtZSI6InBpY29saW5lMjA1OEBnbWFpbC5jb20iLCJlbWFpbCI6InBpY29saW5lMjA1OEBnbWFpbC5jb20iLCJ1c2VyX2lkIjoxLCJuYW1lIjp7ImZvcmVuYW1lIjoiTWF1cmEiLCJzdXJuYW1lIjoiWmFyZW1iYSIsInN1ZmZpeCI6IlBhbiJ9LCJwcm9maWxlIjp7ImFmZmlsaWF0aW9uIjoiQ29ybmVsbCBVbml2ZXJzaXR5IiwiY291bnRyeSI6InVzIiwicmFuayI6Mywic3VibWlzc2lvbl9ncm91cHMiOlsiZ3JwX3BoeXNpY3MiXSwiZGVmYXVsdF9jYXRlZ29yeSI6ImFzdHJvLXBoLkdBIiwiaG9tZXBhZ2VfdXJsIjoiIiwicmVtZW1iZXJfbWUiOnRydWV9LCJ2ZXJpZmllZCI6ZmFsc2V9LCJjbGllbnQiOm51bGwsImVuZF90aW1lIjoiMjAxOS0wMi0xM1QyMjoyMzozNC4yMTY3NzEtMDU6MDAiLCJhdXRob3JpemF0aW9ucyI6eyJjbGFzc2ljIjowLCJlbmRvcnNlbWVudHMiOlsiKi4qIl0sInNjb3BlcyI6WyJwdWJsaWM6cmVhZCIsInN1Ym1pc3Npb246Y3JlYXRlIiwic3VibWlzc2lvbjp1cGRhdGUiLCJzdWJtaXNzaW9uOnJlYWQiLCJ1cGxvYWQ6cmVhZCIsInVwbG9hZDp1cGRhdGUiLCJ1cGxvYWQ6ZGVsZXRlIiwidXBsb2FkOnJlYWRfbG9ncyJdfSwiaXBfYWRkcmVzcyI6bnVsbCwicmVtb3RlX2hvc3QiOm51bGwsIm5vbmNlIjpudWxsfQ.iNOiCGVIZi5iipElLRyUlnx9uucdK7aytjkvr87FTvI
```
Comment (mhl10) [inline README.md]: Can these be example.org addresses?

File: submit/routes/ui.py [hunk 2]
```diff
@@ -0,0 +1,482 @@
...
+        request_data = MultiDict(request.args.items(multi=True))
+    else:
+        request_data = MultiDict(request.form.items(multi=True))
+
+    context = {'pagetitle': title}
+    try:
+        data, code, headers = controller(request.method, request_data,
+                                         request.session, submission_id,
+                                         **kwargs)
+    except (BadRequest, InternalServerError) as e:
```
Comment (mhl10) [inline submit/routes/ui.py]: Screenshot 2019-06-04 15 24 55

I would suggest handling `BadRequest` separately and using a milder generic error message, since this isn't really unexpected and we probably don't want users contacting support in these cases.

Comment (erickpeirson) [reply submit/routes/ui.py]: Most definitely. https://arxiv-org.atlassian.net/browse/ARXIVNG-2196

File: README.md (hunk 1 above)
Comment (erickpeirson) [reply README.md]: They could... the idea with using mimesis is we're generating random (hopefully) fake addresses. Is the worry that these might actually exist?

File: submit/templates/submit/file_upload.html [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+               </span>
+             </a>
+            </div>
+          </div>
+      </div>
+    </div>
+    {% else %}
+    {% if key %}<span class="icon"><i class="fa fa-folder-open-o"></i></span>{{ key }}/{% endif %}
+    <ol class="{% if not key %}file-tree{% endif %}" style="list-style-type: none;">
+      {% for k, subitem in item.items() %}
```
Comment (mhl10) [inline submit/templates/submit/file_upload.html]: Screenshot 2019-06-04 16 17 10

Nit: python's lexicographic sorting feels unnatural for a list of files/folders since it's not how most operating systems (AFAIK) sort files/folder names. I would be inclined to apply `string.casefold()` before sorting.

File: bootstrap.py [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+            scopes.READ_COMPILE,
+            scopes.CREATE_COMPILE
+        ]
+        for user in created_users:
+            token = generate_token(user.user_id, user.email, user.email,
+                                   scope=scope,
+                                   first_name=user.first_name,
+                                   last_name=user.last_name,
+                                   suffix_name=user.suffix_name,
+                                   endorsements=["*.*"])
```
Comment (mhl10) [inline bootstrap.py]: It would be nice to have at least one user with limited/no endorsements. I notice also that the only table related to endorsements in the test DB is `arXiv_endorsement_domains` and it's empty.

File: README.md (hunk 1 above)
Comment (mhl10) [reply README.md]: I was just being über paranoid on a Monday--not with how mimesis is being used at bootstrap time (totally fine) but how the real-looking examples are being commemorated in the README (and indexed, etc.).

File: bootstrap.py (hunk 4 above)
Comment (erickpeirson) [reply bootstrap.py]: Created https://github.com/arXiv/arxiv-submission-ui/issues/90

File: submit/templates/submit/file_upload.html (hunk 3 above)
Comment (erickpeirson) [reply submit/templates/submit/file_upload.html]: Created https://github.com/arXiv/arxiv-submission-ui/issues/83

File: README.md (hunk 1 above)
Comment (erickpeirson) [reply README.md]: Good to be paranoid. Created https://github.com/arXiv/arxiv-submission-ui/issues/92
